AndroidXCI/lib/src/main/kotlin/dev/androidx/ci/testRunner/TestRunnerServiceImpl.kt

```diff
@@ -206,13 +207,57 @@ internal class TestRunnerServiceImpl internal constructor(
         }
     }
 
+    private suspend fun findScreenshotFiles(
+        resultPath: GcsPath,
+        testIdentifiers: List<TestRunnerService.TestIdentifier>
+    ): Map< TestRunnerService.TestIdentifier, List<TestRunnerService.TestCaseArtifact>> {
+        val screenshotArtifactsBlobs = mutableMapOf<TestRunnerService.TestIdentifier, MutableList<TestRunnerService.TestCaseArtifact>>()
+        val screenshotArtifacts: Map<TestRunnerService.TestIdentifier, List<TestRunnerService.TestCaseArtifact>>
+        val testNames = testIdentifiers.associateBy { testIdentifier ->
+            "${testIdentifier.className}_${testIdentifier.name}"
+        }
+        val testRunNumber = testIdentifiers.first().runNumber
+        fun BlobVisitor.fullDeviceId() = relativePath.substringBefore('/', "")
+        googleCloudApi.walkEntires(
+            gcsPath = resultPath
+        ).forEach { visitor ->
+            val runNumber = DeviceRun.create(visitor.fullDeviceId()).runNumber
+            if (runNumber == testRunNumber) {
+                val testName = testNames.keys.find { testName ->
+                    visitor.fileName.startsWith(testName)
+                }
+                val testIdentifier = testNames[testName]
```

Comment:
does this work fine if `testName` is null?

Reply:
Yes, checking if testIdentifier is null in the next line. So, I believe this should be ok.

Reply:
i was wondering what happens with `testNames[null]`.
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/-map/get.html
`get` is declared to receive `K` which is `nonNull` here, idk the implementation but we are passing a nullable value to a non-null key.. not sure why compiler doesn't complain about this :/. do we have a test that covers `testName` being null here?